### PR TITLE
Fixing Button text ignores breakline (\n) #2967

### LINKED ISSUE #2967: Button text ignores breakline (\n) (fixed in `master`)
If I set a text with a breakline like "Hello\nthere" and set this text to the Button text and tooltip, the breakline works on the tooltip but is ignored on the text.

### PR DESCRIPTION
**Pull request by cesarpinho**
After a long review of the code, the discussion on the issue [#2967](https://github.com/godotengine/godot/issues/2967) page and the other merged issues ([#4327](https://github.com/godotengine/godot/issues/4327) and [#21334](https://github.com/godotengine/godot/issues/21334)), I present a solution proposal.

In this [comment](https://github.com/godotengine/godot/issues/2967#issuecomment-164590141), @StraToN say :
> line breaks in Buttons' captions won't work for the moment. I will leave this apart.

And that's what I did. I fix the tabs in the Buttons' captions, Labels text and Control tooltips and fix the line breaks in the Label text and Control tooltip.

**Before**

||  | Break Line (\n) | Tab (\t)
-----|------|--------------|----------------
Inspector | Label | prints the characters |prints the characters
|| Tooltip | prints the characters | prints the characters
|| Button | prints the characters | prints the characters
GDScript | Label | It works | discards the characters
|| Tooltip | It works | discards the characters
|| Button | discards the characters | discards the characters

**After**

||  | Break Line (\n) | Tab (\t)
-----|------|--------------|----------------
Inspector | Label | it works |it works
|| Tooltip | it works | it works
|| Button | prints the characters | it works
GDScript | Label | It works | it works
|| Tooltip | It works | it works
|| Button | discards the characters | it works

Fix  [#2967](https://github.com/godotengine/godot/issues/2967)

### CODE REVIEW

File: scene/gui/control.cpp [hunk 1]
```diff
@@ -41,6 +41,10 @@
 #include "scene/gui/label.h"
 #include "scene/gui/panel.h"
 #include "scene/scene_string_names.h"
+
```
Comment (StraToN) [inline scene/gui/control.cpp]: These lines can be deleted.

**Review by StraToN — DISMISSED** [review]

File: scene/gui/control.cpp (hunk 1 above)
Comment (cesarpinho) [reply scene/gui/control.cpp]: Sorry I forgot to clean it.

**Comment by Anutrix** [comment]
@cesarpinho Please fix the conflicts.

File: core/ustring.cpp [hunk 2]
```diff
@@ -4214,6 +4214,45 @@ String String::unquote() const {
...
+	String s = *this;
+	String slash = "\\";
+	String tab = "\t";
+	size_t tab_pos = s.find(tab, 0);
+
+	while (tab_pos != npos) {
+		String str_begin = s.substr(0, tab_pos);
+		String str_end = s.substr(tab_pos + 1, s.length());
+
+		s = str_begin + "    " + str_end;
```
Comment (dreamsComeTrue) [inline core/ustring.cpp]: Should we always assume, that \t == 4 spaces? Shouldn't it be configurable per usecase?

**Comment by Anutrix** [comment]
Requesting rebase.

**Comment by aaronfranke** [comment]
@cesarpinho The merges you pushed aren't the same as a rebase. Godot requires contributors to make their PRs have good Git history in order to be accepted.

Please see this article for more information: https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html

**Comment by Anutrix** [comment]
@cesarpinho Requesting rebase and squash.

**Comment by aaronfranke** [comment]
This PR has not been in a good state for over a year, closing.

**Pull request closed without merging**